### DISCUSSION TITLE
Regex Support

### DISCUSSION THREAD

**Discussion by PrivOci** [vincjo/datatables#54] [Q&A]
Does the search functionality in the datatables support regular expressions (regex)? If so, could you provide some examples of how to use regex within the search box? If not, are there any workarounds or alternative methods to achieve similar results?

**Comment by vincjo** [comment]
There is currently no support to search entries using a regexp. Would need to add a comparator fn in `search()` args. (will probably come with the next release 👍). 

In the meantime, a possible workaround is to use `filter( value, field, comparator )` instead of `search`. 
`filter` can work with the complete row object like `search`, and allows you to add a custom comparator function. 
Unfortunately I can't test it right now, but it would look like this:

```svelte
<script>
    export let handler // DataHandler instance

    let regex

    // custom comparator:
    const check = (entry, regex) => {
         return entry.match(regex) ? true : false
    } 

    // search workaround:
    const search = () => {
        handler.filter( 
            regex,        
            (row) => row,  // apply on full row object
            check 
        ) 
    } 
</script>

<input 
     type="text" 
     bind:value={regex}
     on:input={search}
/>
```

**Reply by PrivOci** [reply]
Thank you
